realm/realm-library/src/main/cpp/io_realm_internal_OsMap.cpp

```diff
@@ -27,29 +27,136 @@
+        return reinterpret_cast<jlong>(new object_store::Dictionary(shared_realm, obj, ColKey(column_key)));
+    }
+    CATCH_STD()
+    return reinterpret_cast<jlong>(nullptr);
+}
+
+JNIEXPORT jobject JNICALL
+Java_io_realm_internal_OsMap_nativeGetValue(JNIEnv* env, jclass, jlong map_ptr,
+                                            jstring j_key) {
+    try {
+        auto &dictionary = *reinterpret_cast<realm::object_store::Dictionary*>(map_ptr);
+        JStringAccessor key(env, j_key);
+        JavaAccessorContext context(env);
+        return any_cast<jobject>(dictionary.get(context, StringData(key)));
+    }
+    CATCH_STD()
+
+    return nullptr;
+}
+
+JNIEXPORT void JNICALL
+Java_io_realm_internal_OsMap_nativePutMixed(JNIEnv* env, jclass, jlong map_ptr, jstring j_key,
+                                            jlong mixed_ptr) {
+    try {
+        auto &dictionary = *reinterpret_cast<realm::object_store::Dictionary*>(map_ptr);
+        auto &mixed = *reinterpret_cast<Mixed*>(mixed_ptr);
+        JStringAccessor key(env, j_key);
+
+        switch (mixed.get_type()) {
+            case type_Int:
+                dictionary.insert(StringData(key).data(), mixed.get_int());
+                break;
+            case type_Bool:
+                dictionary.insert(StringData(key).data(), mixed.get_bool());
+                break;
+            case type_String:
+                dictionary.insert(StringData(key).data(), mixed.get_string());
+                break;
+            case type_Binary:
+                dictionary.insert(StringData(key).data(), mixed.get_binary());
+                break;
+            case type_Timestamp:
+                dictionary.insert(StringData(key).data(), mixed.get_timestamp());
+                break;
+            case type_Float:
+                dictionary.insert(StringData(key).data(), mixed.get_float());
+                break;
+            case type_Double:
+                dictionary.insert(StringData(key).data(), mixed.get_double());
+                break;
+            case type_Decimal:
+                dictionary.insert(StringData(key).data(), mixed.get_decimal());
+                break;
+            case type_Link:
+                dictionary.insert(StringData(key).data(), mixed.get_link());
+                break;
+            case type_ObjectId:
+                dictionary.insert(StringData(key).data(), mixed.get_object_id());
+                break;
+            case type_UUID:
+                dictionary.insert(StringData(key).data(), mixed.get_uuid());
+                break;
+            case type_TypedLink:
+            case type_LinkList:
+            case type_OldTable:
+            case type_Mixed:
+            case type_OldDateTime:
+                throw std::logic_error(util::format("Invalid data type used for mixed: %1", mixed.get_type()));
+        }
+    }
+    CATCH_STD()
+}
+
+JNIEXPORT void JNICALL
+Java_io_realm_internal_OsMap_nativePutBoolean(JNIEnv *env, jclass, jlong map_ptr,
```

Comment:
Is this needed anymore?

Reply:
I started adding support for mixed and boolean types, so yes. I will also have to add methods for all other supported types.